Samples/auth/Outlook-Add-in-SSO-events/public/commands/commands.js

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+
+const urlOrigin = 'https://localhost:3000'; // Change this if deploying to a different location.
+
+Office.onReady((info) => {
+    if (info.host === Office.HostType.Outlook) {
+        // All console logs from events go to a runtime logging file. For more information, see https://learn.microsoft.com/office/dev/add-ins/testing/runtime-logging
+        console.log('initializing ...' + info);
+    }
+});
+
+// Default SSO settings for acquiring access tokens.
+const defaultSSO = {
+    allowSignInPrompt: false,
+    allowConsentPrompt: false,
+    //    forMSGraphAccess: true, // Leave commented during development testing (sideload) or you get a 13012 error from getAccessToken.
+};
+
+/**
+ * Handle the OnNewMessageCompose event by calling getUserProfile.
+ * This appends a signature with the user's profile to the message body on send.
+ *
+ * @param {Office.AddinCommands.Event} event The OnNewMessageCompose event object.
+ */
+function onItemComposeHandler(event) {
+    callWebServerAPI('GET', urlOrigin + '/getuserprofile')
+        .then((jsonResponse) => {
+            let signature = `${jsonResponse.displayName} \n ${jsonResponse.mail}`;
+            if (jsonResponse.jobTitle !== null) {
+                signature += `\n ${jsonResponse.jobTitle}`;
+            }
+            if (jsonResponse.mobilePhone !== null) {
+                signature += `\n ${jsonResponse.mobilePhone}`;
+            }
+            appendTextOnSend(signature);
```

Comment:
If this calls an async API, it should take in the `event` object and call `event.completed()` after all the work has completed.
So, generally, `event.completed()` should be the last call in each path of the chain, otherwise, we are just relying on luck :)

Reply:
@kwame-ansong, this code calls `event.completed()` in the .then, and .catch. Is there somewhere else I should also call it? I thought that would handle all paths.

Reply:
I see it, the appendTextOnSend has callbacks. I fixed this to call event.completed().